Draft: feat(vpc): doc for GA (#1806)
scaleway/docs-content: network/vpc/how-to/activate-dhcp.mdx

---
meta:
  title: How to activate DHCP on a legacy Private Network
  description: This page explains how to activate DHCP on a legacy Private Network
content:
  h1: How to activate DHCP on a legacy Private Network
  paragraph: This page explains how to activate DHCP on a legacy Private Network
tags: regional-private-network private-network vpc virtual-private-cloud dhcp activate old migrate migration
dates:
  validation: 2023-07-06
  posted: 2023-07-06
categories:
  - network
---

> All Private Networks created after July 12 2023 have **DHCP built into them**. A CIDR block is defined when you create the Private Network. When a resource is attached to the Private Network, an IPv4 and IPv6 address is staticallly allocated to them from this block. This is all managed by our internal [IPAM](/newtork/vpc/concepts/#ipam), and removes the need for manual configuration of your IP addresses.

Reviewer: ```suggestion
All Private Networks created after July 12 2023 have **DHCP built into them**. A CIDR block is defined when you create the Private Network. When a resource is attached to the Private Network, an IPv4 and IPv6 address is statically allocated to them from this block. This is all managed by our internal [IPAM](/newtork/vpc/concepts/#ipam), and removes the need for manual configuration of your IP addresses.
```

However, while DHCP is built into all new Private Networks, it may not be automatically activated for older Private Networks. Previously, DHCP was a function of Public Gateways. Only legacy Private Networks that were previously attached to a Public Gateway with DHCP, have DHCP automatically activated on them. On all other Private Networks created prior to July 12 2023, DHCP remains deactivated by default.

<Message type="tip">
Check out our [migration](/network/vpc/reference-content/vpc-migration/) documentation for full information about how "old" Private Networks were migrated when VPC went into General Availability, and for more information about DHCP.
</Message>

We highly recommend activating DHCP on all of your Private Networks, to let us take care of IP address management in our fully-integrated ecosystem. This document, therefore, explains how to activate DHCP on Private Networks where it has not been automatically activated. 

<Message type="important">
**Activating DHCP is permanent, and there is no way to undo the activation.** <br/><br/>

> Activating DHCP will overwrite any existing manually configued static IP addresses for reosurces attached to the network. Attached resources will receive new dynamically-assigned private IP addresses (one IPv4 and one IPv6) from the network's pre-defined CIDR block. The CIDR block will be displayed at the time of activation (see step 5 below), and you will be able to [view your resources' newly assigned IP addresses](/network/vpc/how-to/attach-resources-to-pn/#how-to-view-the-resource's-ip-address) after activation.

Reviewer: ```suggestion
Activating DHCP will overwrite any existing manually configured static IP addresses for resources attached to the network. Attached resources will receive new dynamically-assigned private IP addresses (one IPv4 and one IPv6) from the network's pre-defined CIDR block. The CIDR block will be displayed at the time of activation (see step 5 below), and you will be able to [view your resources' newly assigned IP addresses](/network/vpc/how-to/attach-resources-to-pn/#how-to-view-the-resource's-ip-address) after activation.
```

</Message>

<Macro id="iam-requirements" />
 
<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com).
  - You have a Private Network created prior to July 12 2023 on which DHCP is not yet activated.
</Message>

## How to activate DHCP on a Private Network

1. Click **VPC** in the **Network** section of the side menu.

    <Lightbox src="scaleway-vpc.webp" alt="" />

>     TODO: check screenshot

Reviewer 2: dont forget :)

2. Click the VPC for the region which corresponds to your Private Network. 

3. Click the **Private Networks** tab.

>     A list of your Private Networks in this VPC displays. Private Networks which do not have DHCP activate have an *!** icon next to them.

Reviewer 2: ```suggestion
    A list of your Private Networks in this VPC displays. Private Networks which do not have DHCP activated have an **!** icon next to them.
```
